### DISCUSSION TITLE
Erro na hora da execução

### DISCUSSION THREAD

**Discussion by lamartinrepereira** [Joaopeuko/Mql5-Python-Integration#1] [General]
Quando executo o example.py   ele me apresenta um erro .Pode me ajudar nisso ?

![image](https://user-images.githubusercontent.com/67126363/104320547-3f28ec80-54c1-11eb-8ead-ffdd100411a2.png)

**Comment by Joaopeuko** [comment]
Olá, tudo bem? 

Obrigado pela mensagem.
Você chegou a modificar alguma coisa no código de exemplo ou ele está exatamente igual?

Você poderia me enviar por favor como está o objeto trade no exemple, por favor?
```
trade = Trade('Example',  # Expert name
              0.1,  # Expert Version
              'EURUSD',  # symbol
              567,  # Magic number
              0.01,  # lot
              30,  # stop loss
              5,  # emergency stop loss
              30,  # take profit
              5,  # emergency take profit
              '9:15',  # It is allowed to trade after that hour. Do not use zeros, like: 09
              '17:30',  # It is not allowed to trade after that hour but let open all the position already opened.
              '17:50',  # It closes all the position opened. Do not use zeros, like: 09
              0.5,  # average fee
              )
```

Acredito que o erro esteja nos valores fornecidos. Alem disso, o Stop Loss e o Take Profit não estão funcionando para Forex no momento, estou trabalhando para corrigir isso. Você terá de usar no Emergency Stop Loss e Emergency Take Profit no lugar.

**Comment by Joaopeuko** [comment]
`"Alem disso, o Stop Loss e o Take Profit não estão funcionando para Forex no momento, estou trabalhando para corrigir isso. Você terá de usar no Emergency Stop Loss e Emergency Take Profit no lugar."`

O problema já foi corrigido.

**Reply by lamartinrepereira** [reply]
Ele só abre uma ordem por vez. Você consegue colocar para abrir mais de uma por vez?

**Reply by Joaopeuko** [reply]
Ele realmente abre apenas uma operação por vez por segurança.

É possível alterar arquivo trade.py para deixar o código mais personalizado para sua necessidade, um bom começo é alterar a linha 178, é onde está a logica que checa a existência de uma operação aberta.

Você pode manter a linha 178 inalterada comentar (#) a linha 132 e 157:
 ``` "position": (Mt5.positions_get()[0].ticket if len(Mt5.positions_get()) == 1 else 0)```
Eu acredito que seja o que você procura. É possível que você tenha alguns erros, principalmente o mencionado acima sobre o StopLoss e TakeProfit.

**Reply by lamartinrepereira** [reply]
fazer algo tipo um implementação das ordens para abrir novas + 1 ou definir o total de ordens >= ,tem como ?

**Reply by Joaopeuko** [reply]
Realmente,

O StopLoss e o TakeProfit não funcionam neste caso.

Você tem de usar o Emergency StopLoss e o Emergency TakeProfit.

Na sua estratégia, você pode deixar os valores para StopLoss e TakeProfit em 50 e o Emergency StopLoss e Emergency TakeProfit em 40.

**Reply by lamartinrepereira** [reply]
Seria nessa parte para escolher entrar para próxima vela? ( sem ser a vela do sinal)

import MetaTrader5 as Mt5


class Rates:
    def __init__(self, symbol, time_frame, start_pos, period):

        self.time = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['time']
        self.open = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['open']
        self.high = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['high']
        self.low = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['low']
        self.close = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['close']
        self.tick_volume = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['tick_volume']
        self.spread = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['spread']
        self.real_volume = Mt5.copy_rates_from_pos(symbol, time_frame, start_pos, period)['real_volume']

**Reply by Joaopeuko** [reply]
Olá, para escolher entrar no próximo candle é preciso escrever o código para isso, no fórum Metatrader5 é possível encontrar alguns exemplos, todos muito bons. 

Acredito que a documentação Python-MT5 poderá te ajudar, assim como o fórum.
https://www.mql5.com/en/docs/integration/python_metatrader5

**Reply by lamartinrepereira** [reply]
A questão de abertura de ordem de compra e venda em pares . Tem como abrir duas  ou mais ordens no mesmo par  ? Percebi que ela só abre outra ordem quando a anterior fecha no mesmo par. Agradeço se puder ajudar nisso

**Reply by Joaopeuko** [reply]
Olá, você pode alterar a linha 178 do arquivo trade.py para abrir mais de uma posição.

`if (len(Mt5.positions_get(symbol=self.symbol)) == 0) and self.trading_time():`

Porem, assim que a sua condição de entrada for atingida ele entrará sucesivamente na operação. Você precisará gerenciar quantas posições você já possui em aberto, e em qual condição você poderá abrir mais de uma posição.
